### ISSUE TITLE
Don't flag naming styles of Unity messages

### ISSUE THREAD

**Issue by Rabadash8820** [microsoft/Microsoft.Unity.Analyzers#404]
#### Problem statement

Unity message methods like `Awake()`, `Start()`, `Update()`, etc. are case-sensitive and must be PascalCase. However, if developers have .editorconfig settings that enforce different naming styles for private methods (such as being camelCase rather than PascalCase), then they may get noisy [IDE1006:Naming Styles](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/naming-rules#rule-id-ide1006-naming-rule-violation) warnings/messages.

For example, with the following `.editorconfig` settings:

```txt
dotnet_naming_style.camel_case.capitalization = camel_case
dotnet_naming_symbols.private_methods.applicable_kinds = method
dotnet_naming_symbols.private_methods.applicable_accessibilities = private
dotnet_naming_rule.private_methods.symbols = private_methods
dotnet_naming_rule.private_methods.style = camel_case
dotnet_naming_rule.private_methods.severity = warning
```

then this code will give a bunch of distracting IDE1006 warnings:

```cs
class MyClass
{
    // All of these declarations would have IDE1006 warnings
    private void Start() { }
    private void Update() { }
}
```

#### Proposed solution

It would be nice if Microsoft.Unity.Analyzers simply suppressed IDE1006 for Unity message methods, similar to how it suppresses those methods from being flagged as unused.

**Comment by Rabadash8820** [comment]
@sailro Thank you so much for implementing this! Do you have any idea when the next NuGet/extension update will ship with these changes?
